src/adapters/uni-v3/adapter.ts

```diff
@@ -0,0 +1,125 @@
+import { Decimal } from '../../utils/numerics';
+import {
+  DecodedOrder,
+  DecodedStrategy,
+  EncodedStrategy,
+} from '../../common/types';
+import { UniV3CastStrategy, UniV3Pool, UniV3Position } from './types';
+import { decodeStrategy } from '../../strategy-management/utils';
+import { encodeOrder } from '../../utils/encoders';
+
+/**
+ * Constants for Uniswap V3 calculations
+ */
+const TICK_BASE = new Decimal('1.0001');
+const DEFAULT_TICK_SPACING = 1;
+
+/**
+ * Calculates the implied Uniswap V3 tick for a given rate
+ * @param {Decimal} rate - The rate to convert to a tick
+ * @param {boolean} roundUp - Whether to round up (ceiling) or down (floor)
+ * @returns {number} The calculated tick value
+ */
+function calculateImpliedTick(rate: Decimal, roundUp: boolean): number {
+  const logRate = rate.ln();
```

Comment:
Can pass `sign: number` to the function, then use `rate.pow(sign).ln()` here.
Then, in function `calculatePositionInformation`, pass `sign = +1` or `sign = -1`.